### ISSUE TITLE
DirWatch New Trunk Recorder Missing System Field

### ISSUE THREAD

**Issue by adorsett** [chuot/rdio-scanner#280]
Upgraded to 6.6.2 and built a new config as I am converting from call-upload to DirWatch. Tried adding the DirWatch in admin and it continued to show red !. Didn’t see an option to select System. Changed to “Default”, selected System, changed back to Trunk Recorder, red ! cleared and I could save config.

**Comment by chuot** [comment]
Just to make sure I understand, you're using a trunk recorder with a rdio scanner. You were initially using an upload script to feed rdio scanner, but now you want to convert to dirwatch. This will only work with dirwatch type trunk-recorder as it will ingest metadata from the json file that comes with the audio file. The filename itself does not contain enough information about the system id. It may work with dirwatch type default, but yes, you have to specify the system id manually because the #SYS metatag won't work for the dirwatch.mask, because there is no such information in the filename, and you would also be stuck with a bunch of json files because the audio file itself would be deleted after ingestion, not the json file with type default.

If you could post screenshots of the definition of dirwatch which gives a red exclamation mark. This will help me understand where the error is.

Personally, I would use the rdio scanner plugin that comes with trunk recorder. It works great and it will be easier to migrate rdio-scanner v7 when it comes out. Unless you have a specific reason not to use it.

**Comment by adorsett** [comment]
so maybe the bug is it’s looking for a System when it shouldn’t be?

Here’s the screen with the red ! That I’ve filled out but can’t save. 
![0C802ED3-9B3C-4CF3-8C1F-8B0A9F0ED6A7](https://user-images.githubusercontent.com/14228398/203795051-10f4022d-5369-4ea0-b78b-d5e2fd892b24.jpeg)

I switched the type to Default which shows more options. 
![52B4E007-44DD-4534-9EAE-E0DEF2B596C0](https://user-images.githubusercontent.com/14228398/203795055-6bfb9cdf-59c6-47ed-9f5a-807cbcf91c68.jpeg)

I set the System
![29F65AC9-6CD8-44B8-81E6-0C56A18E699C](https://user-images.githubusercontent.com/14228398/203795058-34e934d7-3813-4f9a-8dd4-9574b5b3c165.jpeg)

I changed the type back to Trunk Recorder and the red ! is gone and I can save. 
![101060D4-3D0A-4141-97E0-EC82E824DAB8](https://user-images.githubusercontent.com/14228398/203795060-fcdfaa6b-9571-434c-9a24-f4b0c72dfeb6.jpeg)

I took your advice and switched to the plug-in. I liked that DirWatch will import recordings that happened while Rdio Scanner is stopped. I’ve had problems with it where it stops working or whatnot and I lose recordings. They are all sitting there in the folders but the upload script only tries once. It’s after a period of time I notice there’s 29GB of audio clips that haven’t uploaded and I start digging into why. It would be nice if the plug-in would push any files in the archive every time TR starts or periodically retry.
